### PR TITLE
MGMT-11312: Expose equinix metadata in ES

### PR DESCRIPTION
**Pull request by adriengentil**
Retrieve equinix metadata in order to feed it in ES along the existing
job index.

- We now retrieve the labels set by Prow in order to determine the test name, I called it "context" (as a bonus we also retrieve the PR number)
- Then we use the "context" to retrieve the equinix metadata from different possible places as the gather step is not the same depending on the workflow that is used (sno vs. ztp vs. baremetal). It could be inferred from the test name, but I prefer to not make assumption in case we rename them.

### CODE REVIEW

File: src/prowjobsscraper/equinix.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+from prowjobsscraper import utils
+from prowjobsscraper.prowjob import EquinixMetadata, ProwJob, ProwJobs
+
+logger = logging.getLogger(__name__)
+
+
+class EquinixExtractor:
+    """
+    EquinixExtractor parses the Equinix metadata generated by the gather step in the assisted-baremetal workflow.
```
Comment (osherdp) [inline src/prowjobsscraper/equinix.py:14]: I see you're handling ``assisted-baremetal-operator-workflow`` as well, no?

File: src/prowjobsscraper/equinix.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        ],
+        "packet-sno": [
+            "{}/artifacts/{}/baremetalds-sno-gather/artifacts/equinix-metadata.json"
+        ],
+    }
+
+    def __init__(self, client: storage.Client):
+        self._client = client
+
+    def hydrate(self, jobs: ProwJobs) -> None:
```
Comment (osherdp) [inline src/prowjobsscraper/equinix.py:31]: what's your intention by the verb ``hydrate``?
is it like enriching the existing data with more info?

File: tests/prowjobsscraper/test_equinix.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    jobs.items[
+        0
+    ].metadata.labels.cloudClusterProfile = "not-equinix-cloud-cluster-profile"
+
+    storage_client = MagicMock()
+    equinix = EquinixExtractor(storage_client)
+    equinix.hydrate(jobs)
+
+    storage_client.bucket.assert_not_called()
+    assert jobs.items[0].equinixMetadata == None
```
Comment (osherdp) [inline tests/prowjobsscraper/test_equinix.py:114]:
```suggestion
    assert jobs.items[0].equinixMetadata is None
```

**Review by osherdp — CHANGES_REQUESTED** [review]
Looks good!
Added just a bunch of nit comments

File: src/prowjobsscraper/equinix.py (hunk 1 above)
Comment (adriengentil) [reply src/prowjobsscraper/equinix.py:14]: :man_facepalming:

File: src/prowjobsscraper/equinix.py (hunk 2 above)
Comment (adriengentil) [reply src/prowjobsscraper/equinix.py:31]: yes, that is the idea

**Comment by adriengentil** [comment]
For the record, this is how it looks like in Kibana:
image

**Review by osherdp — APPROVED** [review]

**Pull request merged**